File: HelloWorld/procedural_2.rb
#!/usr/bin/env ruby
# encoding: utf-8

# Array.size() is not procedural: it's a method of the object Array.
if ARGV.size != 0 then
  name = ARGV[ 0 ]
end
name ||= 'World'
puts "Hello, #{ name }!"

puts '» José Carlos Monteiro «'

File: HelloWorld/procedural_2.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# https://www.python.org/dev/peps/pep-0263/

# Must import module "sys" to handle command line arguments ( which is not procedural ).
import sys

if __name__ == '__main__' :

  name = 'World' if len( sys.argv ) - 1 == 0 else sys.argv[ 1 ]
  print( 'Hello, %(name)s!' % { 'name' : name } )

  print( '» José Carlos Monteiro «' )
